exonum/src/proto/mod.rs

```diff
@@ -277,6 +279,24 @@ impl ProtobufConvert for Vec<u8> {
     }
 }
 
+// According to protobuf specification only simple scalar types (not floats) and strings can be used
+// as a map keys.
+impl<K, T> ProtobufConvert for HashMap<K, T>
+where
+    K: Eq + std::hash::Hash + Clone,
+    T: ProtobufConvert,
+{
+    type ProtoStruct = HashMap<K, T::ProtoStruct>;
+    fn to_pb(&self) -> Self::ProtoStruct {
+        self.iter().map(|(k, v)| (k.clone(), v.to_pb())).collect()
+    }
+    fn from_pb(mut pb: Self::ProtoStruct) -> Result<Self, failure::Error> {
+        pb.drain()
+            .map(|(k, v)| ProtobufConvert::from_pb(v).map(|v| (k, v)))
+            .collect::<Result<HashMap<_, _>, _>>()
+    }
+}
+
```

Comment:
Let's make clippy happy ? 😄  
```suggestion
impl<K, T, S> ProtobufConvert for HashMap<K, T, S>
where
    K: Eq + Clone + ::std::hash::Hash,
    S: Default + ::std::hash::BuildHasher,
    T: ProtobufConvert,
{
    type ProtoStruct = HashMap<K, T::ProtoStruct>;
    fn to_pb(&self) -> Self::ProtoStruct {
        self.iter().map(|(k, v)| (k.clone(), v.to_pb())).collect()
    }
    fn from_pb(mut pb: Self::ProtoStruct) -> Result<Self, failure::Error> {
        pb.drain()
            .map(|(k, v)| ProtobufConvert::from_pb(v).map(|v| (k, v)))
            .collect::<Result<HashMap<_, _, _>, _>>()
    }
}
```